examples/Consumer/Program.cs

```diff
@@ -62,15 +63,25 @@ public static void Run_Consume(string brokerList, List<string> topics, Cancellat
                 .SetStatisticsHandler((_, json) => Console.WriteLine($"Statistics: {json}"))
                 .SetPartitionsAssignedHandler((c, partitions) =>
```

Comment:
I like it because it protects the user against themselves. downsides: 1. `SetPartitionsIncrementallyAssignedHandler` or `SetPartitionsAssignedIncrementalHandler` or similar seem pretty awkward (better name?). This is the main thing that's detracting from it for me atm 2. The current API matches the Java API better. Not that we've let that stop us before.

I feel inclined to just leave it, but i like your thinking. I'll procrastinate on this for a bit.